backend/app/api/v1/endpoints/auth.py

```diff
@@ -0,0 +1,343 @@
+    verification_result = verify_token(email, v_token)
+
+    if verification_result:
+        user = await db.execute(select(User).filter(User.email == email))
+        user_instance = user.scalar_one()
+        user_instance.is_active = True
+        await db.commit()
+        log.info(f"Email {email} verified successfully")
+    else:
+        log.warning(f"Failed to verify email: {email}")
+
+    return templates.TemplateResponse(
+        "verification_result.html",
+        {"request": request, "verification_result": verification_result},
+    )
+
+
+@router.get(
+    "/user/{id}",
+    status_code=status.HTTP_200_OK,
+    response_model=UserInResponse,
+)
+async def get_user(
+    id: int,
+    db: AsyncSession = Depends(get_db),
+    user_id: int = Depends(get_current_user),
+    admin: str = Depends(get_current_user_role),
+):
+    try:
+        user = await fetch_data_by_id(db, User, id)
+        permission = await check_user_permission(user_id, admin, id)
+        if permission:
+            log.success(f"User with id {id} fetched successfully")
+            return user
+
+    except HTTPException:
+        raise
+
+    except Exception as e:
+        log.error(f"Failed to get user: {e}")
+        raise HTTPException(
+            status_code=status.HTTP_500_INTERNAL_SERVER_ERROR,
+            detail=f"Failed to get user: {str(e)}",
+        )
+
+
+@router.put(
+    "/user/{id}",
+    status_code=status.HTTP_200_OK,
+    response_model=UserInResponse,
+)
+async def update_user(
+    id: int,
+    username: str = Form(None),
+    first_name: str = Form(None),
+    last_name: str = Form(None),
+    contact_number: str = Form(None),
+    db: AsyncSession = Depends(get_db),
+    user_id: int = Depends(get_current_user),
+):
+    log.info(f"Attempting to update user with id: {id}")
+    try:
+        user = await fetch_data_by_id(db, User, id)
+        if not user:
+            log.warning(f"User with id {id} does not exist")
+            raise HTTPException(
+                status_code=status.HTTP_404_NOT_FOUND,
+                detail=f"User with id: {id} does not exist",
+            )
+
+        await check_authorization(user_id, user)
+
+        if username is not None and username != user.username:
+            existing_username = await db.execute(
+                select(User).where(User.username == username).where(User.id != id)
+            )
+            if existing_username.scalar():
+                log.warning(f"Username {username} already exists")
+                raise HTTPException(
+                    status_code=status.HTTP_400_BAD_REQUEST,
+                    detail="Username already exists",
+                )
+            user.username = username
+
+        if first_name is not None:
+            user.first_name = first_name
+
+        if last_name is not None:
+            user.last_name = last_name
+
+        if contact_number is not None:
+            user.contact_number = contact_number
```

Comment:
for such scenarios use the following code block.

```
for attr, value in [('first_name', first_name), ('last_name', last_name), ('contact_number', contact_number)]:
    if value is not None:
        setattr(user, attr, value)

```
```
def update_user_attribute(user, attr_name, attr_value):
    if attr_value is not None:
        setattr(user, attr_name, attr_value)

# Usage
update_user_attribute(user, 'first_name', "John")
update_user_attribute(user, 'last_name', "Doe")
update_user_attribute(user, 'contact_number', "1234567890")
```

You could also use function to such operation. using this many if else is not good practice.

Reply:
Sent data to the update method using predefined schema,
```
user_update_data = {
            "username": username,
            "first_name": first_name,
            "last_name": last_name,
            "contact_number": contact_number,
        }
        
```